Siren/Siren.swift

```
@@ -280,32 +284,44 @@ public class Siren: NSObject {
         
         // Perform Request
         let session = NSURLSession.sharedSession()
-        let task = session.dataTaskWithRequest(request, completionHandler: { (data, response, error) -> Void in
+        let task = session.dataTaskWithRequest(request, completionHandler: { [unowned self] (data, response, error) in
             
             if let error = error {
+
+                self.delegate?.sirenDidFailVersionCheck?(error)
+
                 if self.debugEnabled {
                     print("[Siren] Error retrieving App Store data as an error was returned: \(error.localizedDescription)")
                 }
+
             } else {
+
                 guard let data = data else {
+                    self.delegate?.sirenDidFailVersionCheck?(nil)
```

Here we should return an error that says "no data was returned". Then below we can do `if self.debugEnabled { print(error) }`.